test/utils/checklist.spec.js

```diff
@@ -0,0 +1,113 @@
+import ChecklistUtil from "../../app/utils/checklist";
+import Constants from "../../app/constants/constants";
+
+describe('utils', () => {
+  describe('ChecklistUtil', () => {
+    describe('findAssetLanguagesAndDependencies', () => {
+      it('should return empty result when asset is null or undefined', () => {
+        expect(ChecklistUtil.findAssetLanguagesAndDependencies(null)).toEqual({
+          languages: [],
+          dependencies: [],
+        });
+        expect(ChecklistUtil.findAssetLanguagesAndDependencies(undefined)).toEqual({
+          languages: [],
+          dependencies: [],
+        });
+      });
+
+      it('should return correct languages and dependencies for valid code assets', () => {
+        const languages = {'R': ['r', 'rmd', 'rnw', 'snw'], 'Python': ['py', 'py3', 'pyi'], 'SAS': ['sas'], 'Stata': ['do', 'ado', 'mata'], 'HTML': ['htm', 'html']};
+        Object.keys(languages).forEach((lang) => {
+          languages[lang].forEach((ext) => {
+            expect(ChecklistUtil.findAssetLanguagesAndDependencies({
+              type: Constants.AssetType.FILE,
+              contentTypes: [Constants.AssetContentType.CODE],
+              uri: `path/to/file.${ext}`,
+            })).toEqual({
+              languages: [lang],
+              dependencies: [],
+            });
+          });
+        });
+      });
+
+      it('should return empty result for non-code assets (data, documentation)', () => {
+        expect(ChecklistUtil.findAssetLanguagesAndDependencies({
+          type: Constants.AssetType.FILE,
+          contentTypes: [Constants.AssetContentType.DATA],
+          uri: 'path/to/file.csv',
+        })).toEqual({
+          languages: [],
+          dependencies: [],
+        });
+      });
```

Comment:
I like this - I am wondering if we could have another test similar to this one which returns empty results for non-code assets, but we give the asset a file extension that we would expect it to be flagged if it were a file.  E.g., the content type is still DATA, but we name the uri `path/to/file.py`.  This further ensures that both file extension and content type are being used in the decision.

Reply:
ooh, nice one, don't know why i did not do this

Reply:
actually, right now, we DO NOT make decisions using both the parameters. only the content type is being used, so for all files having CODE content type will pass through it.
but, i'll add one more check in the function, making it robust.